src/lua/datetime.lua

```diff
@@ -420,6 +625,221 @@ local function datetime_tostring(self)
+    local dt = local_dt(self)
+    local secs, nsec = local_secs(self), self.nsec
+    local offset = self.tzoffset
+
+    local is_ym_updated = false
+    adjust = adjust or DEF_DT_ADJUST
+
+    if years ~= 0 then
+        check_range(years, MIN_DATE_YEAR, MAX_DATE_YEAR, 'years')
+        local approx_dt = dt + direction * years * AVERAGE_DAYS_YEAR
+        check_dt(approx_dt, direction, years, 'years')
+        -- tnt_dt_add_years() not handle properly DT_SNAP or DT_LIMIT mode
+        -- so use tnt_dt_add_months() as a work-around
+        dt = builtin.tnt_dt_add_months(dt, direction * years * 12, adjust)
+        is_ym_updated = true
+    end
+    if months ~= 0 then
+        local new_dt = dt + direction * months * AVERAGE_DAYS_MONTH
+        check_dt(new_dt, direction, months, 'months')
+        dt = builtin.tnt_dt_add_months(dt, direction * months, adjust)
+        is_ym_updated = true
+    end
+    if is_ym_updated then
+        secs = (dt - DAYS_EPOCH_OFFSET) * SECS_PER_DAY + secs % SECS_PER_DAY
+    end
+
+    if seconds ~= 0 then
+        secs = secs + direction * seconds
+    end
+
+    if nanoseconds ~= 0 then
+        nsec = nsec + direction * nanoseconds
+    end
+
+    secs, self.nsec = normalize_nsec(secs, nsec)
+    check_dt((secs + SECS_EPOCH_OFFSET) / SECS_PER_DAY, direction)
+    self.epoch = utc_secs(secs, offset)
+    return self
+end
+
+local function date_first(lhs, rhs)
+    if is_datetime(lhs) then
+        return lhs, rhs
+    else
+        return rhs, lhs
+    end
+end
+
+local function error_incompatible(name)
+    error(("datetime:%s() - incompatible type of arguments"):
+          format(name), 3)
+end
+
+--[[
+Matrix of subtraction operands eligibility and their result type
+
+|                 | datetime | interval | table    |
++-----------------+----------+----------+----------+
+| datetime        | interval | datetime | datetime |
+| interval        |          | interval | interval |
+| table           |          |          |          |
+]]
+local function datetime_interval_sub(lhs, rhs)
+    check_date_interval(lhs, "operator -")
+    check_date_interval_table(rhs, "operator -")
+    local left_interval = is_table(lhs) or is_interval(lhs)
+    local right_interval = is_table(rhs) or is_interval(rhs)
+    local year, month, secs, nsec, adjust
+
+    -- left is date, right is interval
+    if not left_interval and right_interval then
+        year, month, secs, nsec, adjust = interval_decouple_args(rhs)
+        return datetime_increment_by(datetime_new_copy(lhs), -1, year, month,
+                                     secs, nsec, adjust)
+    -- left is date, right is date
+    elseif not left_interval and not right_interval then
+        local obj = interval_new()
+        obj.sec, obj.nsec = normalize_nsec(lhs.epoch - rhs.epoch,
+                                           lhs.nsec - rhs.nsec)
+        return obj
+    -- both left and right are intervals
+    elseif left_interval and right_interval then
```

Comment:
That comment isn't really revealing anything new considering what is written right below it. I suggest to drop this one and the one above about `left is date, right is date`. It is obvious.

Reply:
On 1 side I agreed with you, on the other side - there were already 2 branches before with not so much obvious (at least from 1st sight) cases, i.e. `left is date, right is interval` or `left is date, right is date`. And I'd prefer to keep this comment in the 3rd branch also. To make this looking more consistent. 

![image](https://user-images.githubusercontent.com/4956130/145257176-c2d2a3a2-40d0-49bc-ad36-d1947febee60.png)

[And sometimes I _do_ care about colors and how it looks like in editor]

Reply:
IMO, all these 3 comments are obvious and useless. But ok.